From 28377b3f54c01c85cf0c771fb71d5a4c3391b0e1 Mon Sep 17 00:00:00 2001
From: Cedric Bondier <cedricx.bondier@intel.com>
Date: Mon, 7 Jan 2013 16:07:34 +0100
Subject: [PORT FROM MAIN][BT] Multivendor support: adding libbt-vendor conf for x86 targets

BZ: 76413 72228 99491 103267 112157 117525 103648 117478 129034 140721 140739

Including the specific Intel configuration file for all x86 targets
in libbt-vendor configuration.

Some compile-time configuration variables for Bluetooth are board
specific.

As a consequence, the Bluedroid configuration files are created in
product device specific configuration folders.

Note: FwPatchFilePath is configured from bt_vendor.conf when possible and
actually needed. It is configured at runtime through android properties when the
platform variant is not sufficient.

Change-Id: I9e34bf523a15277e4f73ee2c636164dc5cacfeb6
Orig-Change-Id: Ia7de3b9049b0586cea0b8c24b4239a66af203a6e
Category: device enablement
Domain: CWS.BT-Common
Origin: internal
Upstream-Candidate: no, proprietary
Orig-Change-Id: I0fb33f2839448fc5423212c67116919cce6189c9
Orig-Change-Id: I3c16072332ac6cfb321839f15b5c271d7c5b259b
Orig-Change-Id: I362c3c538c9d8db1c1319790c3d54bc3115fb2c6
Orig-Change-Id: I9cb2be4731913afcd2dd1f5526cad346bcb76e9f
Orig-Change-Id: If0ee232e3970c863174fec3239a49aac5c9d2eac
Signed-off-by: Xavier Boniface <xavierx.boniface@intel.com>
Signed-off-by: Patrick Combes <patrick.combes@intel.com>
Signed-off-by: Raffaele Aquilone <raquiloX>
Signed-off-by: Julien Gros <julien.gros@intel.com>
Signed-off-by: Celine Chotard <celine.chotard@intel.com>
Signed-off-by: Laurent FERT <laurent.fert@intel.com>
Signed-off-by: Mathieu Perrier <mathieux.perrier@intel.com>
Signed-off-by: Eric Seibel <eric.seibel@intel.com>

Reviewed-by: Bondier, CedricX <cedricx.bondier@intel.com>
Reviewed-by: Aquilone, RaffaeleX <raffaelex.aquilone@intel.com>
Reviewed-by: Biselli, MirkoX <mirkox.biselli@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Isaksen, FrodeX <frodex.isaksen@intel.com>
Reviewed-by: Combes, Patrick <patrick.combes@intel.com>
Reviewed-by: Vamour, Cedric <cedric.vamour@intel.com>
Reviewed-by: Fert, Laurent <laurent.fert@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Reviewed-by: Boniface, XavierX <xavierx.boniface@intel.com>
Reviewed-by: Molero Rodenas, BeatrizX <beatrizx.molero.rodenas@intel.com>
Reviewed-by: Tinelli, Vincent <vincent.tinelli@intel.com>
Reviewed-by: Dixon, CharlesX <charlesx.dixon@intel.com>
Reviewed-by: Fert, Laurent <laurent.fert@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Tested-by: Tinelli, Vincent <vincent.tinelli@intel.com>
Tested-by: Dixon, CharlesX <charlesx.dixon@intel.com>
Reviewed-on: http://android.intel.com:8080/123799
Reviewed-on: http://android.intel.com:8080/123807
Reviewed-on: http://android.intel.com:8080/126532
Reviewed-on: http://android.intel.com:8080/126153
Reviewed-on: http://android.intel.com:8080/126531
Reviewed-on: http://android.intel.com:8080/133967
Reviewed-by: Poignet, Cedric <cedric.poignet@intel.com>
---
 conf/intel/baylake/bt_vendor.conf     |    5 +++++
 conf/intel/byt_m_crb/bt_vendor.conf   |    5 +++++
 conf/intel/byt_t_ffrd8/bt_vendor.conf |    5 +++++
 conf/intel/ctp7160/bt_vendor.conf     |   24 ++++++++++++++++++++++++
 conf/intel/redhookbay/bt_vendor.conf  |    4 ++++
 conf/intel/saltbay/bt_vendor.conf     |    2 ++
 6 files changed, 45 insertions(+), 0 deletions(-)
 create mode 100644 conf/intel/baylake/bt_vendor.conf
 create mode 100644 conf/intel/byt_m_crb/bt_vendor.conf
 create mode 100644 conf/intel/byt_t_ffrd8/bt_vendor.conf
 create mode 100644 conf/intel/ctp7160/bt_vendor.conf
 create mode 100644 conf/intel/redhookbay/bt_vendor.conf
 create mode 100644 conf/intel/saltbay/bt_vendor.conf

diff --git a/conf/intel/baylake/bt_vendor.conf b/conf/intel/baylake/bt_vendor.conf
new file mode 100644
index 0000000..e23e986
--- /dev/null
+++ b/conf/intel/baylake/bt_vendor.conf
@@ -0,0 +1,5 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/ttyMFD0
+
+# Firmware patch file location
+FwPatchFilePath = /etc/firmware/bt/vv/
diff --git a/conf/intel/byt_m_crb/bt_vendor.conf b/conf/intel/byt_m_crb/bt_vendor.conf
new file mode 100644
index 0000000..e23e986
--- /dev/null
+++ b/conf/intel/byt_m_crb/bt_vendor.conf
@@ -0,0 +1,5 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/ttyMFD0
+
+# Firmware patch file location
+FwPatchFilePath = /etc/firmware/bt/vv/
diff --git a/conf/intel/byt_t_ffrd8/bt_vendor.conf b/conf/intel/byt_t_ffrd8/bt_vendor.conf
new file mode 100644
index 0000000..e23e986
--- /dev/null
+++ b/conf/intel/byt_t_ffrd8/bt_vendor.conf
@@ -0,0 +1,5 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/ttyMFD0
+
+# Firmware patch file location
+FwPatchFilePath = /etc/firmware/bt/vv/
diff --git a/conf/intel/ctp7160/bt_vendor.conf b/conf/intel/ctp7160/bt_vendor.conf
new file mode 100644
index 0000000..8efdd81
--- /dev/null
+++ b/conf/intel/ctp7160/bt_vendor.conf
@@ -0,0 +1,24 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/ttyMFD0
+
+# Firmware patch file location
+FwPatchFilePath = /etc/firmware/bt/vv/
+
+LPM_IDLE_TIMEOUT_MULTIPLE = 5
+SCO_USE_I2S_INTERFACE = FALSE
+SCO_I2SPCM_IF_ROLE = 1
+SCO_I2SPCM_IF_SAMPLE_RATE = 0
+SCO_I2SPCM_IF_CLOCK_RATE = 0
+SCO_PCM_IF_CLOCK_RATE = 0
+SCO_PCM_IF_FRAME_TYPE = 1
+SCO_PCM_IF_SYNC_MODE = 1
+SCO_PCM_IF_CLOCK_MODE = 1
+PCM_DATA_FMT_SHIFT_MODE = 0
+PCM_DATA_FMT_FILL_BITS = 0
+PCM_DATA_FMT_FILL_METHOD = 3
+PCM_DATA_FMT_FILL_NUM = 3
+PCM_DATA_FMT_JUSTIFY_MODE = 0
+BTVND_DBG = FALSE
+BTHW_DBG = FALSE
+VNDUSERIAL_DBG = FALSE
+UPIO_DBG = FALSE
diff --git a/conf/intel/redhookbay/bt_vendor.conf b/conf/intel/redhookbay/bt_vendor.conf
new file mode 100644
index 0000000..6a5ebf3
--- /dev/null
+++ b/conf/intel/redhookbay/bt_vendor.conf
@@ -0,0 +1,4 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/hci_tty
+
+# Note: for TI chips, the firmware file location is not set here
diff --git a/conf/intel/saltbay/bt_vendor.conf b/conf/intel/saltbay/bt_vendor.conf
new file mode 100644
index 0000000..e6d8154
--- /dev/null
+++ b/conf/intel/saltbay/bt_vendor.conf
@@ -0,0 +1,2 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/ttyMFD0
-- 
1.7.4.1

